### ISSUE TITLE
Extension does not update cookie when it is already set

### ISSUE THREAD

**Issue by CommanderRedYT** [tubearchivist/browser-extension#49]
I have a old cookie that is blocked by youtube. When I delete it and re-sync the cookie, it works again (also works without cookie). But it seems the extension only updates it when it is empty.

**Comment by ajw107** [comment]
I have the same problem, tubearchivist not getting new videos due to an issue with the cookie it is using (I assume too old).  Luckily there is an even easier workaround, you don;t need to delete the old cookie, in the extension just untick Option -> Sync Youtube Cookies, and then tick it again.  Saves having to scroll through tubearchivist's settings until the issue is fixed.

**Comment by bbilly1** [comment]
Yeah, this probably shouldn't have been a checkbox, a button would have been more appropriate. 

The goal of that functionality is to copy the cookie on demand. Currently that will happen, once the checkbox is set to true.

Permanent sync would be another solution but will have potential sideffects, e.g. if you switch Google accounts.

I think first step is to change the UI and make this a button instead. If you want to look into, please do.

**Comment by amanofcharacter** [comment]
@ajw107 @bbilly1 Is unchecking and rechecking still a fix for this? I'm having the same issue, but unchecking/rechecking is not allowing me to download videos again. I've even cleared my YouTube cookie out of my browser before, during, and after this process to see if I can force create a new YouTube cookie. Whatever I do, TA will not download. Am I missing something about the how the YouTube cookie and the TA cookie sync works?

**Comment by CommanderRedYT** [comment]
@amanofcharacter My solution was to also always use chrome and deactivate cookie sync for Firefox. With that it worked

**Comment by amanofcharacter** [comment]
@CommanderRedYT Thanks, I'll try that out

**Comment by amanofcharacter** [comment]
@CommanderRedYT Hm, that didn't seem to work for me. So I was using FireFox before. Here's what I tried:
 
- Downloaded and installed Chrome
- Unchecked the FireFox cookie sync in the TA extebsion
- Logged into YouTube on chrome
- Added the TA extension and synced the cookie

I still get downloads that attempt to start, but only make it about 10% and then stop. Then they seem to start again, but also stop again. My current error for most videos is the "ERROR: unable to download video data: HTTP Error 403: Forbidden".

Prior to trying out the VPN method, the error was telling me to sign in to confirm I'm not a bot.

**Comment by bbilly1** [comment]
Please stay on topic: This issue is about how the checkbox "Sync youtube cookies" really should be a button called "Copy YouTube cookies" that does that as advertised on click. 

If you want to help implement that, please open a PR for that.

An no, using cookies is not a guarantee that YT is not blocking you, as using cookies makes every of your requests personally identifiable and attributable to you and your account. So that makes things easier for YT to block you.

**Comment by amanofcharacter** [comment]
Id call it on topic…

My issue has since been resolved. The sync checkbox was very _apparently_ not adding/updating the cookie. At this point in time when I encountered what seemed to be the same issue, on/around 1/16/2025, TA was not down long for a lot of people, related to an update to YT-dlp. I’d get the 403 error, but also the “sign in” error. Before I found additional discussion around YT-dlp update, fixable by manual update, I observed that unchecking/re-checking _appeared_ to update the cookie, but download behavior remained _unchanged_. Errors regarding “sign in” _remained_ as this error. That’s how I got here, given the thread title.

Only after updating YT-dlp could I download again. At that point I did not have my cookie syncing. In a couple hours, boom, “sign in error” again. Hit sync, problem solved, unlike before the update, where it wasn’t solved.

So, whether the need for YT-dlp update was _causing_ the cookie not to sync, or just was coincidence? I don’t know, I can’t say. But that was the verifiable visible behavior.

I’m totally new to TA, only got it a week ago.

**Comment by Kozi801** [comment]
I am getting the same issue. Did someone get a solution? I am already to the latest version fo yt-dlp. Is there a specific command to update yt-dlp within tubearchivist correctly.

**Comment by bbilly1** [comment]
There is an attempt of "continuous sync" here: https://github.com/tubearchivist/browser-extension/tree/release-TA-v0.5.0, only for your unstable builds. That introduces a onChange listener for the cookies on YT, and sync the cookies when something changes. It's quite aggressive, but so far effective.

**Comment by bbilly1** [comment]
OK, continuous cookie sync cookies is shipped with v0.4.0, please update.
